server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -128,34 +115,57 @@ protected ClusterState execute(ClusterState currentState, AddColumnRequest reque
         return currentState;
     }
 
-    static AddColumnRequest addMissingParentColumns(AddColumnRequest request, DocTableInfo currentTable) {
+
+    public record NormalizedRequest (AddColumnRequest request, boolean allExist) {}
+
+    /**
+     * Removes already existing columns which can be created by another request in case of concurrent dynamic mapping updates.
+     * We should still iterate over all existing columns to run the type validation but we don't need any further handling of them.
+     * Adds missing parent columns to the leaf columns so that it's possible to build and traverse the tree.
+     */
+    static NormalizedRequest normalizeColumns(AddColumnRequest request, DocTableInfo currentTable) {
         List<Reference> newColumns = new ArrayList<>();
-        boolean anyMissing = false;
-        for (var newColumn : request.references()) {
```

Comment:
removed anyMissing as I had a bug when anyMissing is false but I had to copy request anyway since we now have 2 sources of updates.